state_version.go

```diff
@@ -55,18 +57,25 @@ type StateVersionList struct {
 
 // StateVersion represents a Terraform Enterprise state version.
 type StateVersion struct {
-	ID                 string                   `jsonapi:"primary,state-versions"`
-	CreatedAt          time.Time                `jsonapi:"attr,created-at,iso8601"`
-	DownloadURL        string                   `jsonapi:"attr,hosted-state-download-url"`
-	Serial             int64                    `jsonapi:"attr,serial"`
-	VCSCommitSHA       string                   `jsonapi:"attr,vcs-commit-sha"`
-	VCSCommitURL       string                   `jsonapi:"attr,vcs-commit-url"`
-	ResourcesProcessed bool                     `jsonapi:"attr,resources-processed"`
-	StateVersion       int                      `jsonapi:"attr,state-version"`
-	TerraformVersion   string                   `jsonapi:"attr,terraform-version"`
-	Modules            *StateVersionModules     `jsonapi:"attr,modules"`
-	Providers          *StateVersionProviders   `jsonapi:"attr,providers"`
-	Resources          []*StateVersionResources `jsonapi:"attr,resources"`
+	ID           string    `jsonapi:"primary,state-versions"`
+	CreatedAt    time.Time `jsonapi:"attr,created-at,iso8601"`
+	DownloadURL  string    `jsonapi:"attr,hosted-state-download-url"`
+	Serial       int64     `jsonapi:"attr,serial"`
+	VCSCommitSHA string    `jsonapi:"attr,vcs-commit-sha"`
+	VCSCommitURL string    `jsonapi:"attr,vcs-commit-url"`
+	// Whether Terraform Cloud has finished populating any StateVersion fields that required async processing.
+	// If `false`, some fields may appear empty even if they should actually contain data; see comments on
```

Comment:
Am I understanding correctly that if `ResourcesProcessed`  is set to `false`, then all the fields listed beneath (StateVersion, TerraformVersion, Modules, etc) will not have values populated yet?

Reply:
Yes, that's correct. I think a nuance though is that `ResourcesProcessed` could be false, and then when it's true there could still be no values for those fields (depending on the state).

Reply:
Unfortunately, _some of them might while others will not._ There are _up to two_ rounds of async processing that might happen on a stateversion, and the atlas API as of today doesn't have granular progress tracking on them -- `resources-processed` gets flipped after the _final_ round of async processing completes, so it's the best signal we have that everything's done.

Reply:
Gotcha 👍